core/src/filed/backup.cc

```diff
@@ -1406,6 +1411,11 @@ static inline bool SendPlainData(b_ctx& bctx)
     bctx.jcr->ReadBytes += bytes_read; /* count bytes read */
   }
   sd->msg = bctx.msgsave; /* restore read buffer */
+
+  if (read_error) {
+    // the following code recognizes read errors
+    sd->message_length = -1;
```

Comment:
um... aren't negative message lengths signals? If that's the case here maybe `BNET_EOD` would show the intent better.

Reply:
Its not supposed to be a signal.  Its just how the code communicates sadly :(  I did not want to add a major change now.  This should get changed in a coming pr.

Reply:
See here:
```
  if (sd->message_length < 0) { /* error */
    BErrNo be;
    Jmsg(jcr, M_ERROR, 0, _("Read error on file %s. ERR=%s\n"), ff_pkt->fname,
         be.bstrerror(ff_pkt->bfd.BErrNo));
    if (jcr->JobErrors++ > 1000) { /* insanity check */
      Jmsg(jcr, M_FATAL, 0, _("Too many errors. JobErrors=%d.\n"),
           jcr->JobErrors);
    }
[...]
```

Reply:
I added the missing second half of that comment.